lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb

```diff
@@ -676,6 +676,15 @@ def convert_date(date)
       end
 
       def convert_date_no_day(date)
+        year = date[0, 4]
+        month = date[5, 2]
+        day = date[8, 2]
+
+        # somehow, partial dates with the 'XX' (i.e. "2020-01-XX or 2020-XX-XX") are getting past FE validation
+        # fix here in the backend while a proper FE solution is found
+        return year if month == 'XX'
+        return "#{year}-#{month}" if day == 'XX'
```

Comment:
should we normalize these "XX"s - either `.downcase` or `.upcase` the `month`/`day` variables?

Reply:
@aurora-a-k-a-lightning  thanks! I incorporated Kyle's suggestion and the upcase one. 

@christinec-fftc I think I'll hold off on accounting for the XXXX year for now, unless we actually start seeing this. If a date is XXXX-XX-XX, I would think (hope) it's coming in as nil or not at all. Otherwise there could be things like XXXX-01-XX or XXXX-XX-01 which again seem like they would be bizarre and rare. I may have to eat these word down the line, but for now I'd like to keep it simple and target the known scenario.